src/main/java/pl/gacik/coordinates/Coordinates2DInterface.java

```diff
@@ -0,0 +1,56 @@
+package pl.gacik.coordinates;
+
+public interface Coordinates2DInterface extends Comparable {
```

Comment:
It's ONLY my opinion, but for me it's better to have just 'Coordinates2D' name for interface, because now you are injecting this interface(implementation) with name 'Coordinates2DInterface' everywhere (example in method 'addPair' in CrossBoard class). More often, when people have only one implementation of an interface, they just add 'Impl' as suffix to name (I saw a lot of commercial project with this kind of naming). 

Example: Coordinates2D(it's interface injecting in a lot of parts of your code), Coordinates2DImpl(implementation of Coordinates2D)

It's just my suggestion :)